### PR TITLE
Feat: pruneBranches isolates part of the dom tree, and removes all other parts in events

### PR DESCRIPTION
**Pull request by Juice10**
This is super useful for debugging huge events.

Syntax:
```js
const ids = [150]; // list of ids who's parents and children we'd like to keep
const prunedEvents = pruneBranches(events, {keep: ids});
```

### CODE REVIEW

File: packages/rrweb-cutter/src/index.ts [hunk 1]
```diff
@@ -96,6 +98,132 @@ function cutEvents(
...
+            ...event,
+            data: {
+              ...event.data,
+              positions: prunedPositions,
+            },
+          } as eventWithTime);
+      } else if ('id' in event.data) {
+        if (treeSet.has(event.data.id)) result.push(event);
+      } else if (event.data.source === IncrementalSource.Mutation) {
+        const { removes, adds, texts, attributes } = event.data;
```
Comment (Mark-Fenng) [inline packages/rrweb-cutter/src/index.ts:151]: For the adds mutation, I come up with one corner situation: if one node is moved to be a child of another pruned node, will this cause errors during the playback?

**Review by Mark-Fenng — COMMENTED** [review]
Love this feature! It will really help us with debugging. The implementation looks good and I only have one question about a corner case I'm not so sure.

File: packages/rrweb-cutter/src/index.ts (hunk 1 above)
Comment (Juice10) [reply packages/rrweb-cutter/src/index.ts:151]: Hmm interesting (edge) case. Hadn't thought of that one yet

Comment (Juice10) [reply packages/rrweb-cutter/src/index.ts:177]: Done: https://github.com/rrweb-io/rrweb/pull/1011/commits/57f6ad6fe8f1a8886490cc19fee04577ef01f555

Comment (Juice10) [reply packages/rrweb-cutter/src/index.ts:177]: Thanks for the suggestion @Mark-Fenng

**Comment by Juice10** [comment]
@Mark-Fenng I think this one is ready to be merged into your `trim` branch. You cool with me doing that?

**Review by Mark-Fenng — APPROVED** [review]

**Pull request merged**